ADAL/src/utils/NSString+ADURLExtensions.m

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "NSString+ADURLExtensions.h"
+
+@implementation NSString (ADURLExtensions)
+
+- (NSString *)adAuthorityWithCloudInstanceName:(NSString *)cloudInstanceName
+{
+    if (!cloudInstanceName)
+    {
+        return self;
+    }
+    
+    NSURLComponents *urlComponents = [NSURLComponents componentsWithString:self];
+    NSString *loginHost = [NSString stringWithFormat:@"login.%@", cloudInstanceName];
```

Comment:
In the near future it will return full name including login, e.g "login.microsoftonline.de" (in a different parameter called cloud_instance_dns_name). So, we won't need to construct loginHost ourselves, but we can just directly swap hosts. Basically it will kill just this one line.